### PR TITLE
[MOV] web_editor, knowledge, mass_mailing, website: editor lib to src

### PR DESCRIPTION
**Pull request by Zynton**
Odoo-Editor, having originally been developed with the purpose of being a standalone library, was put into a `lib` folder. Since in the meantime the library plans were aborted and since its presence outside the `src` directory makes translations unavailable for it, this moves the editor to `src/js/editor/odoo-editor`.

task-2880723

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by dmo-odoo** [comment]
cc @mart-e

File: addons/web/tooling/_eslintrc.json [hunk 1]
```diff
@@ -37,6 +37,12 @@
...
     "py": "readonly",
     "ClipboardJS": "readonly",
-    "globalThis": "readonly"
+    "globalThis": "readonly",
+    "module": "readonly",
+    "chai": "readonly",
+    "describe": "readonly",
+    "it": "readonly",
+    "mocha": "readonly",
+    "DOMPurify": "readonly"
```
Comment (sdegueldre) [inline addons/web/tooling/_eslintrc.json:46]: Confused about this, this is the web tooling's config and I don't see diff in the whitelist in _eslintignore. If you want to use the web tooling in web_editor that's fine but you should probably add the files to the whitelist. If you don't want to use the web tooling and only did this to make runbot green, it's unnecessary, you only need to add them to the test_eslint config.

Comment (dmo-odoo) [reply addons/web/tooling/_eslintrc.json:46]: Good catch. We were quite confused about this. Thanks !

Comment (dmo-odoo) [reply addons/web/tooling/_eslintrc.json:46]: @Zinston The proper list to update is in `addons/test_lint/tests/eslintrc`.

File: addons/web_editor/views/editor.xml [hunk 2]
```diff
@@ -168,4 +168,31 @@
...
+            <script src="/web_editor/static/lib/DOMPurify.js"></script>
+            <script src="/web_editor/static/src/js/editor/odoo-editor/test/lib/mocha.js"></script>
+            <script src="/web_editor/static/src/js/editor/odoo-editor/test/lib/chai.js"></script>
+            <script>mocha.setup('bdd').fullTrace()</script>
+            <script src="/web_editor/static/src/js/editor/odoo-editor/test/editor-test.js" type="module"></script>
+        </t>
+        <div id="mocha"></div>
+        <div id="editor-test-container"></div>
+    </t>
+</template>
```
[the remark this answers is not in the document]
Comment (Zinston) [reply addons/web_editor/views/editor.xml:197]: I didn't remove it because we still use it in the demo page and I think that's still useful to have.

**Pull request closed without merging**